forc-pkg/src/pkg.rs

```diff
@@ -2523,6 +2555,70 @@ pub fn compile(
     }
 }
 
+impl PkgEntry {
+    /// Tries to retrieve the `PkgEntry` as a `PkgTestEntry`, panics otherwise.
+    pub fn expect_test_entry(&self) -> Result<PkgTestEntry> {
```

Comment:
We can let the method user choose whether or not they incur the cost of the `clone`.

```suggestion
    pub fn expect_test_entry(&self) -> Result<&PkgTestEntry> {
```

It's worth keeping in mind, the Rust std sets the convention that `expect` is generally used to indicate that the method will `panic!` if the expected outcome isn't the case (e.g. `Option::expect`, `Result::expect`), whereas here we return a `Result`. 

It might be better to add this method to the `PkgEntryKind` type as that is the only data that is required to check this.

Something like the following might give the method user more flexibility:
```rust
impl PkgEntryKind {
    /// Returns `Some` if the `PkgEntryKind` is `Test`.
    pub fn test(&self) -> Option<PkgEntryTest> {
        match self {
            PkgEntryKind::Test(test) => Some(test),
            _ => None,
        }
    }
}
```
We can still use it to expect a value:
```rust
let test_entry = pkg_entry.kind.test().expect("expected test entry point");
```
or return an error:
```rust
let test_entry = pkg_entry.kind.test().ok_or(anyhow!("expected test entry point"))?;
```
It also removes the need for the `PkgEntry::is_test` method as we can do this:
```rust
pkg_entry.kind.test().is_some()
```

None of these notes are super important, but I just wanted to point out these patterns as they can show up a lot in Rust library design around these sorts of `*Kind` enums!